host_core/lib/host_core/web_assembly/imports.ex

```diff
@@ -599,7 +599,8 @@ defmodule HostCore.WebAssembly.Imports do
   # Load the guest response indicated by the location and length into the :guest_response state field.
   defp guest_response(_api_type, context, agent, ptr, len) do
     memory = context.memory
-    gr = Wasmex.Memory.read_binary(memory, ptr, len)
+    caller = context.caller
+    gr = Wasmex.Memory.read_binary(caller, memory, ptr, len)
```

Comment:
Nit: save a couple lines with `context.caller` and `context.memory` as you've done above?